examples/Exploring-different-models.ipynb

```diff
@@ -0,0 +1,805 @@
+    "\n",
+    "### Learning objectives\n",
+    "- Preparing the data with NVTabular\n",
+    "- Training different deep learning-based recommender models with Merlin Models"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0f354e2a",
+   "metadata": {},
+   "source": [
+    "## Feature Engineering with NVTabular"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9eadd76d",
+   "metadata": {},
+   "source": [
+    "When we work on a new recommender systems, we explore the dataset, first."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "64c4b1d6",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "2022-03-16 15:32:17.079946: I tensorflow/core/common_runtime/gpu/gpu_process_state.cc:214] Using CUDA malloc Async allocator for GPU: 0\n",
+      "2022-03-16 15:32:17.080101: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1525] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 16254 MB memory:  -> device: 0, name: Quadro GV100, pci bus id: 0000:15:00.0, compute capability: 7.0\n"
+     ]
+    }
+   ],
+   "source": [
+    "import os\n",
+    "os.environ[\"TF_GPU_ALLOCATOR\"]=\"cuda_malloc_async\"\n",
+    "import cudf\n",
+    "import glob\n",
+    "import gc\n",
+    "\n",
+    "import nvtabular as nvt\n",
+    "from nvtabular.ops import *\n",
+    "from example_utils import workflow_fit_transform\n",
+    "from example_utils import save_results\n",
+    "\n",
+    "from merlin.schema.tags import Tags\n",
+    "from merlin.schema import Schema\n",
+    "\n",
+    "import merlin.models.tf as mm\n",
+    "import merlin.models.tf.dataset as tf_dataloader\n",
+    "\n",
+    "from merlin.io.dataset import Dataset\n",
+    "from merlin.schema.io.tensorflow_metadata import TensorflowMetadata\n",
+    "from merlin.models.tf.blocks.core.aggregation import CosineSimilarity\n",
+    "\n",
+    "import tensorflow as tf"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2f39ad9e",
+   "metadata": {},
+   "source": [
+    "First, we define our input and output paths."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "a6e0eaaa",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "DATA_FOLDER = \"/workspace/data/\"\n",
+    "train_path = os.path.join(DATA_FOLDER, 'train/*.parquet')\n",
+    "test_path = os.path.join(DATA_FOLDER, 'test/*.parquet')\n",
+    "output_path = '/workspace/data/processed'"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ebe79531",
+   "metadata": {},
+   "source": [
+    "NVTabular has already implemented multiple calculations, called `ops`. An `op` can be applied to a `ColumnGroup` from an overloaded `>>` operator\n",
+    "**Example:**<br>\n",
+    "```python\n",
+    "features = [ column_name, ...] >> op1 >> op2 >> ...\n",
+    "```\n",
+    "\n",
+    "This may sounds more complicated as it is. Let's define our the pipeline."
```

Comment:
Should we explain NVTabular + showing the NVTabular workflow.

1) We explain NVTabular in the NVTabular repo
2) We explain it in the 2nd notebook: Applying to your own dataset

Should we have a utils function for NVTabular workflow + returning merlin.io.Dataset ?
Should we use merlin.ioDataset instead of loading the schema + using the tf_dataloader.BatchedDataset

Reply:
Yes, we can remove NVTabular explanations. good point.

`Should we use merlin.ioDataset instead of loading the schema + using the tf_dataloader.BatchedDataset` --> this means we dump all to `model.fit()` and it makes model.fit() too crowded.. I prefer it as lean as existing tf.keras examples.